en/components/grid_search.md

```diff
@@ -0,0 +1,327 @@
+
+- The `grid.lastSearchInfo.matchInfoCache.length` value will give us the total results count. 
+- The `grid.lastSearchInfo.activeMatchIndex` value will give us the index position of the current occurrence (match). 
+
+```html
+<!--searchgrid.component.html-->
+
+<div *ngIf="grid.lastSearchInfo">
+    <span *ngIf="grid.lastSearchInfo.matchInfoCache.length > 0">
+        {{ grid.lastSearchInfo.activeMatchIndex + 1 }} of {{ grid.lastSearchInfo.matchInfoCache.length }} results
+    </span>
+    <span *ngIf="grid.lastSearchInfo.matchInfoCache.length == 0">
+        No results
+    </span>
+</div>
+```
+
+#### Add search buttons
+
+In order to freely search and navigate among our search results, let's create a couple of buttons by invoking the `findNext` and the `findPrev` methods inside the buttons' respective click event handlers.
+
+```html
+<!--searchgrid.component.html-->
+
+<div class="searchButtons">
+    <input type="button" value="Previous" (click)="grid.findPrev(searchText, caseSensitive)" />
+    <input type="button" value="Next" (click)="grid.findNext(searchText, caseSensitive)" />
+</div>
+```
+
+#### Add keyboard search
+
+We can also allow the users to search navigate the results by using the keyboard's arrow keys and the Enter key. In order to achieve this, we can handle the **keydown** event of our search input and invoke the `findNext`/`findPrev` methods depending on which key the user has pressed.
+
+```html
+<!--searchgrid.component.html-->
+
+<input #search1 id="search1" placeholder="Search" [(ngModel)]="searchText" (ngModelChange)="grid.findNext(searchText, caseSensitive)"
+       (keydown)="searchKeyDown($event)" />
+```
+
+```typescript
+// searchgrid.component.ts
+
+public searchKeyDown(ev) {
+    if (ev.key === "Enter" || ev.key === "ArrowDown" || ev.key === "ArrowRight") {
+        this.grid.findNext(this.searchText, this.caseSensitive);
+    } else if (ev.key === "ArrowUp" || ev.key === "ArrowLeft") {
+        this.grid.findPrev(this.searchText, this.caseSensitive);
+    }
+}
+```
+
+#### Case sensitive
+
+Now let's allow the user to choose whether the search should be case sensitive or not. For this purpose we can use a simple checkbox input by binding our **caseSensitive** property to its **checked** property and handle the **change** event by toggling our property and invoking the `findNext` method.
+
+```html
+<!--searchgrid.component.html-->
+
+<span>Case sensitive</span>
+<input type="checkbox" [checked]="caseSensitive" (change)="updateSearch()">
+```
+
+```typescript
+// searchgrid.component.ts
+
+public updateSearch() {
+    this.caseSensitive = !this.caseSensitive;
+    this.grid.findNext(this.searchText, this.caseSensitive);
+}
+```
+
+#### Persistence
+
+What if we would like to filter and sort our grid or even to add and remove records? After such operations, the highlights of our current search automatically update and persist over any text that matches the **searchText**!
```

Comment:
Add one more sentence, along the lines of "Furthermore, the search will work with paging and will persist the highlights through changes of the grid's perPage property."